src/main/java/com/azure/cosmos/kafka/connect/sink/CosmosDBSinkConnector.java

```diff
@@ -48,9 +62,61 @@ public void stop() {
     public ConfigDef config() {
         return CosmosDBSinkConfig.getConfig();
     }
-    
+
     @Override
     public String version() {
         return this.getClass().getPackage().getImplementationVersion();
     }
+
+    @Override
+    public Config validate(Map<String, String> connectorConfigs) {
+        Config config = super.validate(connectorConfigs);
+        if (config.configValues().stream().anyMatch(cv -> !cv.errorMessages().isEmpty())) {
+            return config;
+        }
+
+        Map<String, ConfigValue> configValues = config.configValues().stream().collect(
+            Collectors.toMap(ConfigValue::name, Function.identity()));
+
+        validateConnection(connectorConfigs, configValues);
+        validateTopicMap(connectorConfigs, configValues);
+
+        return config;
+    }
+
+    private void validateConnection(Map<String, String> connectorConfigs, Map<String, ConfigValue> configValues) {
+        String endpoint = connectorConfigs.get(CosmosDBSinkConfig.COSMOS_CONN_ENDPOINT_CONF);
+        String key = connectorConfigs.get(CosmosDBSinkConfig.COSMOS_CONN_KEY_CONF);
+        try {
+            readAllDatabases(endpoint, key);
+        } catch (Exception e) {
+            configValues.get(CosmosDBSinkConfig.COSMOS_CONN_ENDPOINT_CONF)
+                .addErrorMessage("Could not connect to endpoint with error: " + e.getMessage());
+            configValues.get(CosmosDBSinkConfig.COSMOS_CONN_KEY_CONF)
+                .addErrorMessage("Could not connect to endpoint with error: " + e.getMessage());
+        }
+    }
+
+    private void validateTopicMap(Map<String, String> connectorConfigs,
+        Map<String, ConfigValue> configValues) {
+
+        String topicMap = connectorConfigs.get(CosmosDBSinkConfig.COSMOS_CONTAINER_TOPIC_MAP_CONF);
+        try {
+            TopicContainerMap.deserialize(topicMap);
+        } catch (Exception e) {
+            configValues.get(CosmosDBSinkConfig.COSMOS_CONTAINER_TOPIC_MAP_CONF)
+                .addErrorMessage(INVALID_TOPIC_MAP_FORMAT);
+        }
+    }
+
+    // visible for testing
+    CosmosPagedIterable<CosmosDatabaseProperties> readAllDatabases(String endpoint, String key) {
+        return new CosmosClientBuilder()
+            .endpoint(endpoint)
+            .key(key)
+            .userAgentSuffix(CosmosDBConfig.COSMOS_CLIENT_USER_AGENT_SUFFIX + version())
+            .buildClient()
+            .readAllDatabases();
```

Comment:
Would this require any additional permission?

Reply:
From what I can tell, the connector requires the cosmos primary master key to be configured in property `connect.cosmos.master.key` . The primary key will 
> Provide access to accounts, databases, users, and permissions.
Cannot be used to provide granular access to containers and docu

https://docs.microsoft.com/en-us/azure/cosmos-db/database-security#primary-keys

Reply:
correct. the master key would have the needed permissions for execute an operation such as readAllDatabases(). no problem  here.